lib/modules/manager/pub/artifacts.ts

```diff
@@ -0,0 +1,78 @@
+import { TEMPORARY_ERROR } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import {
+  getSiblingFileName,
+  readLocalFile,
+  writeLocalFile,
+} from '../../../util/fs';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+export async function updateArtifacts({
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  if (
+    !(
+      config.updateType === 'lockFileMaintenance' ||
+      config.updateType === 'lockfileUpdate'
+    ) &&
+    (updatedDeps === undefined || updatedDeps.length < 1)
```

Comment:
```suggestion
    (updatedDeps.length < 1)
```
i don't think updatedDeps can be undefined